### ISSUE TITLE
 Unknown address type

### ISSUE THREAD

**Issue by domgolonka** [btcsuite/btcd#1406]
I am trying to connect using the RPCCLIENT on a testnet.  Whenever I call GetNewAddress or GetAccountAddress 
```
address, err := this.client.GetAccountAddress("default")
	if (err != nil) {
		logs.Error(err)
		return false
	}
```

It always shows the err "unknown address type" even though the account exists:
```
#btcctl --rpcuser=xxx --rpcpass=xxx --wallet --testnet --rpcserver=localhost:8332 listaccounts
{
  "default": 0,
  "main": 0,
  "imported": 0
}
```

```
#btcctl --rpcuser=xxx --rpcpass=xxx --wallet  --rpcserver=localhost:8332 getaccountaddress default
mumXAA57R7AeSg2UkZroAHu3SfAUWgnjzA
```

**Comment by domgolonka** [comment]
Found the bug.  It calls btcutil.DecodeAddress(addrStr, &chaincfg.MainNetParams) instead of &chaincfg.TestNet3Params

```
func (r FutureGetAddressesByAccountResult) Receive() ([]btcutil.Address, error) {
	res, err := receiveFuture(r)
	if err != nil {
		return nil, err
	}

	// Unmashal result as an array of string.
	var addrStrings []string
	err = json.Unmarshal(res, &addrStrings)
	if err != nil {
		return nil, err
	}

	addrs := make([]btcutil.Address, 0, len(addrStrings))
	for _, addrStr := range addrStrings {
		addr, err := btcutil.DecodeAddress(addrStr,
			&chaincfg.MainNetParams)
		if err != nil {
			return nil, err
		}
		addrs = append(addrs, addr)
	}

	return addrs, nil
}
```

**Comment by kobayashi5884** [comment]
I have the same issue. This issue seems not to be fixed yet. Is there any workaround now?

**Comment by rusticbison** [comment]
I also just experienced this issue. Would be good to see this re-opened.

**Comment by JFixby** [comment]
https://github.com/btcsuite/btcd/issues/1466

**Comment by Rjected** [comment]
Fixed by #1467

**Comment by jakesyl** [comment]
Closing as per #1467. Feel free to reopen if this is still an issue @domgolonka
